src/planscape/forsys/views.py

```diff
@@ -3,43 +3,321 @@
+  else:
+    min_x = int(np.floor((e[2] - o[0]) / s[0]))
+    max_x = int(np.ceil((e[0] - o[0]) / s[0]))
+
+  if s[1] > 0:
+    min_y = int(np.floor((e[1] - o[1]) / s[1]))
+    max_y = int(np.ceil((e[3] - o[1]) / s[1]))
+  else:
+    min_y = int(np.floor((e[3] - o[1]) / s[1]))
+    max_y = int(np.ceil((e[1] - o[1]) / s[1]))
+
+  data = raster.bands[0].data()
+
+  sum = 0
+  count = 0
+  for y in range(min_y, max_y + 1, 1):
+    for x in range(min_x, max_x + 1, 1):
+      d = data[y][x]
+      if np.isnan(d):
+        continue
+
+      # TODO: adjust these equations for the case where skew != [0, 0]
+      xpoly = o[0] + x*s[0]
+      ypoly = o[1] + y*s[1] 
+      p = Point((xpoly, ypoly))
+      if not p.within(polygon):
+        continue
+
+      sum = sum + d
+      count = count + 1
+
+  if count == 0:
+    return {"mean": 0, "count": 0}
+  return {"mean": sum/count, "count": count}
+
+def transform_into_forsys_df_data(condition_rasters, boundaries, project_area):
+  kConditionPrefix = "cond"
+  kPriorityPrefix = "p"
+  kUnitAreaCost = 5000
+
+  data = {}
+  data['proj_id'] = []
+  data['stand_id'] = []
+  data['shape_name'] = []
+  for c in condition_rasters.keys():
+    data[kConditionPrefix + "_" + c] = []
+    data[kPriorityPrefix + "_" + c] = []
+    data['area'] = []
+    data['cost'] = []
+
+  for b in boundaries:
+    # raster boundary seems to be 80 units off from the huc-12 boundaries.
+    # TODO: adjust raster data.
+    geo = project_area.intersection(b.geometry)
+    geo = translate_polygon(geo, 80, 0)
+    geo.transform(CoordTransform(SpatialReference(4269), SpatialReference(9822)))
+
+    # TODO: set project_area ID from an external source
+    data['proj_id'].append(1)
+    # TODO: double-check that it makes sense to use this ID.
+    data['stand_id'].append(b.id)
+    # TODO: double-check that this field is necessary.
+    data['shape_name'].append(b.shape_name)
+    data['area'].append(geo.area)
+    # TODO: adjust cost as a function of treatment type.
+    data['cost'].append(kUnitAreaCost * geo.area)
+
+    for c in condition_rasters.keys():
+      d = get_condition_data(condition_rasters[c], geo)
+      data[kConditionPrefix + "_" + c].append(d['mean'])
+      # TODO: adjust improvement score as a function of treatment type.
+      data[kPriorityPrefix + "_" + c].append((1.0 - d['mean']) * d['count'])
+
+  return data 
+
+# Runs a forsys scenario sets call.
+def run_forsys_scenario_sets(npdf, priorities):
+  kPriorityPrefix = "p"
+
+  import rpy2.robjects as robjects
+  robjects.r.source(os.path.join(settings.BASE_DIR, 'forsys/scenario_sets.R'))
+  scenario_sets_function_r = robjects.globalenv['scenario_sets']
+ 
+  # TODO: add inputs for thresholds.
+  # TODO: clean-up: pass header names (e.g. proj_id) into scenario_sets_function_r.
+  rdf = convert_dictionary_of_lists_to_rdf(npdf)
+
+  priority_headers = []
+  for p in priorities:
+    priority_headers.append(kPriorityPrefix + "_" + p)
```

Comment:
This is confusing because you hard-coded `kPriorityPrefix = "p"` but here the variable `p` contains the string with the variable name. Since you're hard-coding `"p"`, why not change it to something like `priority_headers.append("p_" + p)` and make this a one-liner?

Reply:
In terms of eng style, it's preferable to use constants rather than direct values.
How the constants are set in this file could be refined.